src/binding/c/rma_api.txt

```diff
@@ -0,0 +1,466 @@
+    operation op applies to elements of that predefined type. target_datatype must
+    not specify overlapping entries, and the target buffer must fit in the target
+    window or in attached memory in a dynamic window.
+    
+    Any of the predefined operations for 'MPI_Reduce', as well as 'MPI_NO_OP' or
+    'MPI_REPLACE' can be specified as op. User-defined functions cannot be used. A
+    new predefined operation, 'MPI_NO_OP', is defined. It corresponds to the
+    associative function f (a, b) = a; i.e., the current value in the target memory
+    is returned in the result buffer at the origin and no operation is performed on
+    the target buffer. 'MPI_NO_OP' can be used only in 'MPI_Get_accumulate',
+    'MPI_Rget_accumulate', and 'MPI_Fetch_and_op'. 'MPI_NO_OP' cannot be used in
+    'MPI_Accumulate', 'MPI_Raccumulate', or collective reduction operations, such as
+    'MPI_Reduce' and others.
+*/
+
+MPI_Put:
+    .desc: Put data into a memory window on a remote process
+    .seealso: MPI_Rput
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Raccumulate:
+    .desc: Accumulate data into the target process using remote memory access
+    .skip: ThreadSafe
+    .seealso: MPI_Accumulate
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+/*
+    Notes:
+    The basic components of both the origin and target datatype must be the same
+    predefined datatype (e.g., all 'MPI_INT' or all 'MPI_DOUBLE_PRECISION').
+*/
+
+MPI_Rget:
+    .desc: Get data from a memory window on a remote process
+    .seealso: MPI_Get
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Rget_accumulate:
+    .desc: Perform an atomic, one-sided read-and-accumulate
+    .skip: ThreadSafe
+    .seealso: MPI_Get_accumulate MPI_Fetch_and_op
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+/*
+    Notes:
+    This operations is atomic with respect to other "accumulate" operations.
+    
+    The get and accumulate steps are executed atomically for each basic element in
+    the datatype (see MPI 3.0 Section 11.7 for details). The predefined operation
+    'MPI_REPLACE' provides fetch-and-set behavior.
+    
+    The basic components of both the origin and target datatype must be the same
+    predefined datatype (e.g., all 'MPI_INT' or all 'MPI_DOUBLE_PRECISION').
+*/
+
+MPI_Rput:
+    .desc: Put data into a memory window on a remote process and return a request
+    .seealso: MPI_Put
+    .impl: mpid
+    .earlyreturn: pt2pt_proc_null
+
+MPI_Win_allocate:
+    .desc: Create and allocate an MPI Window object for one-sided communication
+    .seealso: MPI_Win_allocate_shared MPI_Win_create MPI_Win_create_dynamic MPI_Win_free
+    .impl: mpid
+
+MPI_Win_allocate_shared:
+    .desc: Create an MPI Window object for one-sided communication and shared memory access, and allocate memory at each process
+    .seealso: MPI_Win_allocate MPI_Win_create MPI_Win_create_dynamic MPI_Win_free MPI_Win_shared_query
+    .impl: mpid
```

Comment:
In `win_allocate_shared.c`, error `.N MPI_ERR_INFO` and `.N MPI_ERR_SIZE` are lost. For invalid size argument, I think it is better to return `MPI_ERR_SIZE` but not `MPI_ERR_ARG`. Same for `disp`.

Reply:
The code format in `win_allocate_shared.c` seems weird (see `MPIR_OBJ_PUBLISH_HANDLE ` line)
```
    if (win_ptr) {
    /* Initialize a few fields that have specific defaults */
    win_ptr->name[0] = 0;
    win_ptr->errhandler = 0;
        MPIR_OBJ_PUBLISH_HANDLE(*win, win_ptr->handle);
    }
```

Reply:
I see in many files, the header comment is lost. For example, in `win_allocate_shared.c` there was a comment:
```
This is a collective call executed by all processes in the group of comm. On
each process i, it allocates memory of at least size bytes that is shared among
all processes in comm, and returns...
```

Reply:
> In `win_allocate_shared.c`, error `.N MPI_ERR_INFO` and `.N MPI_ERR_SIZE` are lost. For invalid size argument, I think it is better to return `MPI_ERR_SIZE` but not `MPI_ERR_ARG`. Same for `disp`.
> 

Let me add `MPIR_ERRTEST_RMASIZE` and `MPIR_ERRTEST_DISPUNIT` for them, will update.

Reply:
> 
> 
> The code format in `win_allocate_shared.c` seems weird (see `MPIR_OBJ_PUBLISH_HANDLE ` line)
> 
> ```
>     if (win_ptr) {
>     /* Initialize a few fields that have specific defaults */
>     win_ptr->name[0] = 0;
>     win_ptr->errhandler = 0;
>         MPIR_OBJ_PUBLISH_HANDLE(*win, win_ptr->handle);
>     }
> ```

Added commit to fix python script

Reply:
> 
> 
> I see in many files, the header comment is lost. For example, in `win_allocate_shared.c` there was a comment:
> 
> ```
> This is a collective call executed by all processes in the group of comm. On
> each process i, it allocates memory of at least size bytes that is shared among
> all processes in comm, and returns...
> ```

Oh I see. That notes was before the parameter description block, which my original scraping script didn't anticipate. They really should be after the parameter block as all the other man pages. I'll run a script to check all the missing ones. To be updated.